server/utils/applications/summaryListUtils.ts

```diff
@@ -55,38 +58,80 @@ const taskResponsesAsSummaryListItems = (
   return items
 }
 
-const attachDocumentsSummaryListItems = (
-  application: Application,
-  task: UiTask,
-  pageName: string,
-  showActions: boolean,
-) => {
+/**
+ * Return a list items from a document list for the check-your-answers page
+ * @param documents list of documents to render
+ * @param personCrn from application - used in document download path
+ * @param changePath - if defined, adds a 'change' link
+ * @param otherDocumentDetails - other documents text box contents
+ */
+export const getDocumentSummaryListItems = (
+  documents: Array<Document>,
+  personCrn: string,
+  changePath: string,
+  otherDocumentDetails: string,
+): Array<SummaryListItem> => {
   const items: Array<SummaryListItem> = []
 
-  documentsFromApplication(application).forEach(document => {
+  const getAction = (visuallyHiddenText = null as string) =>
+    changePath
+      ? {
+          actions: {
+            items: [
+              {
+                href: changePath,
+                text: 'Change',
+                ...(visuallyHiddenText ? { visuallyHiddenText } : {}),
+              },
+            ],
+          },
+        }
+      : {}
+
+  documents.forEach(document => {
     const item: SummaryListItem = {
       key: {
-        html: `<a href="/applications/people/${application.person.crn}/documents/${document.id}" data-cy-documentId="${document.id}">${document.fileName}</a>`,
+        html: `<a href="/applications/people/${personCrn}/documents/${document.id}" data-cy-documentId="${document.id}">${document.fileName}</a>`,
       },
-      value: { text: document?.description || '' },
-    }
-    if (showActions) {
-      item.actions = {
-        items: [
-          {
-            href: applyPaths.applications.pages.show({ task: task.id, page: pageName, id: application.id }),
-            text: 'Change',
-            visuallyHiddenText: document.fileName,
-          },
-        ],
-      }
+      value: { text: document?.description },
+      ...getAction(document.fileName),
     }
     items.push(item)
   })
-
+  if (otherDocumentDetails?.length)
+    items.push({
+      key: { text: 'Tell us about any other relevant documents' },
+      value: { text: otherDocumentDetails },
+      ...getAction(),
+    })
   return items
 }
 
+const attachDocumentsSummaryListItems = (
+  application: Application,
+  task: UiTask,
+  pageName: string,
+  showActions: boolean,
+) => {
+  const personCrn: string = application.person.crn
+  const changePath: string = showActions
+    ? applyPaths.applications.pages.show({ task: task.id, page: pageName, id: application.id })
+    : undefined
+  let otherDocumentDetails: string
+  try {
+    otherDocumentDetails = retrieveQuestionResponseFromFormArtifact(
+      application,
+      AttachDocuments,
+      'otherDocumentDetails',
+    ) as string
```

Comment:
Not sure this type casting is useful?

Reply:
True. At one stage I was declaring inside the try/catch.
Heh - I just noticed there is a routine `retrieveOptionalQuestionResponseFromFormArtifact` that absorbs the exception if the value is not populated. That should simplify things a bit.